src/python/pants/backend/python/tasks/resolve_requirements_task_base.py

```diff
@@ -42,11 +55,16 @@ def resolve_requirements(self, req_libs):
 
       interpreter = self.context.products.get_data(PythonInterpreter)
       path = os.path.realpath(os.path.join(self.workdir, str(interpreter.identity), target_set_id))
-
       # Note that we check for the existence of the directory, instead of for invalid_vts,
       # to cover the empty case.
       if not os.path.isdir(path):
         with safe_concurrent_creation(path) as safe_path:
+          # Handle locally-built python distribution dependencies.
+          built_dists = self.context.products.get_data(BuildLocalPythonDistributions.PYTHON_DISTS)
```

Comment:
Just noticed something: What happens if only these local dists change? The invalidated() block won't notice that, so the requirements pex won't be rebuilt. It seems like their originating targets should participate in the invalidation?

Reply:
Oh, I guess this is why the local_dist_targets arg exists? But they aren't passed in practice, as far as I can tell.  And the duplication is a bit weird. Any reason not to do the inject_synthetic_dist_requirements() when the distribution is created, i.e., in `build_local_python_distributions.py`? That seems more idiomatic. Then downstream tasks like this one don't have to distinguish between the different provenances of requirement libraries, and can treat them uniformly.